src/main/java/com/SBS/Services/CouponService.java

```diff
@@ -0,0 +1,52 @@
+package com.SBS.Services;
+
+import com.SBS.Models.Bet;
+import com.SBS.Models.Coupon;
+import com.SBS.Repositories.CouponRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.SBS.Models.CouponStatus.IN_PLAY;
+import static com.SBS.Models.CouponStatus.IN_PROGRESS;
+
+@Service
+public class CouponService {
+    CouponRepository couponRepository;
+
+    CouponService(CouponRepository couponRepository) {
+        this.couponRepository = couponRepository;
+    }
+
+    Optional<Coupon> addBet(Coupon coupon, Bet bet) {
+        if (coupon.getCouponStatus() == IN_PROGRESS) {
```

Comment:
Takie porównaia robi się odwrotnie. Bo jakby tam było equal zamiast == to jest otwarta furta do NullPoitnera, kiedy `getCoupunStatus()` zwróciło by nulla. A tak dobry nawyk bedziesz mial do zaczynania porównań od zdefiniowanych i pewnych rzeczy.
`IN_PROGRESS == coupon.getCouponStatus()`